src/main/java/fluke/Fluke.java

```diff
@@ -0,0 +1,212 @@
+     */
+    public void addTodo(String command) throws FlukeException {
+        String parsedDescription = Parser.parseTodoCommand(command);
+        Task taskAdded = this.tasks.addTodo(parsedDescription);
+        this.ui.showTaskAdded(taskAdded, tasks);
+    }
+
+    /**
+     * Tells Fluke to add a Deadline to its tasks.
+     * @param command Command given to Fluke.
+     * @throws FlukeException if an error occurs while adding the task, for instance, with invalid inputs.
+     */
+    private void addDeadline(String command) throws FlukeException {
+        String[] parsedCommand = Parser.parseDeadlineCommand(command);
+        String description = parsedCommand[0];
+        String byDate = parsedCommand[1];
+        Task taskAdded = this.tasks.addDeadline(description, byDate);
+        this.ui.showTaskAdded(taskAdded, tasks);
+    }
+
+    /**
+     * Tells Fluke to add an Event to its tasks.
+     * @param command Command given to Fluke.
+     * @throws FlukeException if an error occurs while adding the task, for instance, with invalid inputs.
+     */
+    private void addEvent(String command) throws FlukeException {
+        String[] parsedCommand = Parser.parseEventCommand(command);
+        String description = parsedCommand[0];
+        String fromDate = parsedCommand[1];
+        String toDate = parsedCommand[2];
+        Task taskAdded = this.tasks.addEvent(description, fromDate, toDate);
+        this.ui.showTaskAdded(taskAdded, tasks);
+    }
+
+    /**
+     * Tells Fluke to mark a task as done.
+     * @param command Command given to Fluke.
+     * @throws FlukeException if an error occurs while marking a task as done, for instance, with invalid inputs.
+     */
+    private void markTaskAsDone(String command) throws FlukeException {
+        int index = Parser.parseMarkAsDoneCommand(command);
+        Task taskMarked = tasks.markTaskAsDone(index);
+        this.ui.showTaskMarkedAsDone(taskMarked);
+    }
+
+    /**
+     * Tells Fluke to mark a task as not done.
+     * @param command Command given to Fluke.
+     * @throws FlukeException if an error occurs while marking a task as not done, for instance, with invalid inputs.
+     */
+    private void markTaskAsUndone(String command) throws FlukeException {
+        int index = Parser.parseMarkAsUndoneCommand(command);
+        Task taskMarked = tasks.markTaskAsUndone(index);
+        this.ui.showTaskMarkedAsUndone(taskMarked);
+    }
+
+    /**
+     * Tells Fluke to delete a task.
+     * @param command Command given to Fluke.
+     * @throws FlukeException if an error occurs while deleting a task, for instance, with invalid inputs.
+     */
+    private void deleteTask(String command) throws FlukeException {
+        int index = Parser.parseDeleteCommand(command);
+        Task deleted = tasks.deleteTask(index);
+        this.ui.showTaskDeleted(deleted, tasks);
+    }
+
+    /**
+     * Fluke attempts to find tasks with the corresponding keyword.
```

Comment:
Nice use of comments to describe what each method does, maybe possible to add a line between the description and the params to fit the module's code style. 
```suggestion
     * Fluke attempts to find tasks with the corresponding keyword.
     
```
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/40513197/cd7b77e4-b3bb-4eec-a649-41981f23001c)